### PR TITLE
Remove jake (hopefully for real this time)

### PR DESCRIPTION
**Pull request by weswigham**
A couple of our build tasks will need to be updated, too; but I think this should be all that's needed in TS itself.

### CODE REVIEW

**Comment by weswigham** [comment]
Oooooo that feeling when you put `async` functions in the gulpfile but `node 6` doesn't support them natively. @rbuckton got a preference on how to handle that?

**Comment by sandersn** [comment]
I would like to keep jake around until

```sh
> gulp
> gulp
```

Has the second invocation run fast, just like jake does today.

**Comment by weswigham** [comment]
Well...
 Now that's true.

**Comment by weswigham** [comment]
@sandersn See https://github.com/Microsoft/TypeScript/pull/29447 😛

**Comment by rbuckton** [comment]
Do you need to add the change to support using LKG on CI that we added to jake?

**Comment by weswigham** [comment]
I'm doing something a little different - one second, just validating it a bit.

**Comment by weswigham** [comment]
There we go - rather than disabling lkg on CI (which leads to frustrating test diffs locally), I'm now running a sanity check built/local build on CI only after tests pass.

**Comment by sandersn** [comment]
Node 6 is now out in favour of node 11, so that's one more obstacle out of the way.

**Comment by weswigham** [comment]
Alright, looks like everything is now green. Anyone wanna peek over the CI logs to see that you're OK with the now gulp-based ones and give this the :+1:?

File: tests/baselines/reference/api/tsserverlibrary.d.ts [hunk 1]
```diff
@@ -8347,7 +8347,7 @@ declare namespace ts.server {
         excludedFiles: ReadonlyArray<NormalizedPath>;
         private typeAcquisition;
         updateGraph(): boolean;
-        getExcludedFiles(): readonly NormalizedPath[];
+        getExcludedFiles(): ReadonlyArray<NormalizedPath>;
```
Comment (rbuckton) [inline tests/baselines/reference/api/tsserverlibrary.d.ts]: I thought this had already been addressed in #29783?

Comment (weswigham) [reply tests/baselines/reference/api/tsserverlibrary.d.ts]: I un-addressed it to make the nightlies build again.

**Comment by weswigham** [comment]
Alright. All build pipelines are on `gulp`-based scripts and the sync script for VS is setup to use gulp, too. Once this build passes now that I've synced with `master` and resolved the `package.json` conflict, I plan to merge this. That should unblock merging `user` suite PRs verbatim, and make the code coverage pipeline green again. This PR's been up for a week since approval, but if anyone has any final words before this merges, the next hour or so is what's left (barring catastrophic failures requiring a revert).

**Comment by Kingwl** [comment]
Wooo, this scared me.（bye jake）

**Comment by alexeagle** [comment]
@weswigham how do you have incremental builds now?

**Comment by weswigham** [comment]
Our gulpfile is shelling out to `tsc -b` and not using `gulp-typescript` is how. :neutral_face:

**Pull request merged**